CMakeLists.txt

```diff
@@ -390,3 +390,16 @@ if (PHOTON_BUILD_TESTING)
     endif ()
     GenerateStandaloneCTestScript(${CMAKE_SOURCE_DIR} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/CTestTestfile.cmake)
 endif ()
+
+################################################################################
+
+# Install headers and libs
+add_custom_target(copy_includes ALL
+    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_SOURCE_DIR}/include ${CMAKE_BINARY_DIR}/include
+    COMMENT "Copying include files to build directory: ${CMAKE_BINARY_DIR}/include"
+)
+add_dependencies(copy_includes photon_static)
+
+install(DIRECTORY ${CMAKE_BINARY_DIR}/include/ DESTINATION include)
```

Comment:
```suggestion
install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION include)
install(FILES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libphoton.a ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libphoton.so DESTINATION lib)
```

Reply:
Actually we don't need to copy include dir to build dir? So the `copy_includes` target is not necessary. 

And those depentent static libs have been `--whole-archive`ed into libphoton.so, so you can just specify the so file to install.